testing/e2e/clients/acr.go

```diff
@@ -63,3 +77,22 @@ func (a *acr) GetName() string {
 func (a *acr) GetId() string {
 	return a.id
 }
+
+func (a *acr) BuildAndPush(ctx context.Context, imageName string) error {
+	lgr := logger.FromContext(ctx).With("image", imageName, "name", a.name, "resourceGroup", a.resourceGroup, "subscriptionId", a.subscriptionId)
+	ctx = logger.WithContext(ctx, lgr)
+	lgr.Info("starting to build and push image")
+	defer lgr.Info("finished building and pushing image")
+
+	// Ideally, we'd use the sdk to build and push the image but I couldn't get it working.
+	// I matched everything on the az cli but wasn't able to get it working with the sdk.
+	// https://github.com/Azure/azure-cli/blob/5f9a8fa25cc1c980ebe5e034bd419c95a1c578e2/src/azure-cli/azure/cli/command_modules/acr/build.py#L25
+	cmd := exec.Command("az", "acr", "build", "--registry", a.name, "--image", imageName, ".")
+	cmd.Stdout = newLogWriter(lgr, "building and pushing acr image: ", nil)
+	cmd.Stderr = newLogWriter(lgr, "building and pushing acr image: ", to.Ptr(slog.LevelError))
```

Comment:
love to see the `slog` package in the wild

Reply:
And the `slog` package loves to see you

Reply:
May `slog` be with you